go-controller/pkg/util/nicstobridge.go

```diff
@@ -115,36 +112,37 @@ func delAddRoute(oldLink, newLink netlink.Link, route netlink.Route) error {
 }
 
 func saveRoute(oldLink, newLink netlink.Link, routes []netlink.Route) error {
-	for i := range routes {
-		route := routes[i]
-
-		// Handle routes for default gateway later.  This is a special case for
-		// GCE where we have /32 IP addresses and we can't add the default
-		// gateway before the route to the gateway.
-		if route.Dst == nil && route.Gw != nil && route.LinkIndex > 0 {
-			continue
-		} else if route.Dst != nil && !route.Dst.IP.IsGlobalUnicast() {
+	// Handle routes for default gateway later.  This is a special case for
+	// GCE where we have /32 IP addresses and we can't add the default
+	// gateway before the route to the gateway.
+	for _, route := range routes {
+		if route.Dst == nil {
 			continue
 		}
-
-		err := delAddRoute(oldLink, newLink, route)
-		if err != nil {
-			return err
+		// only consider routes for /32 or /128 IP addresses
+		ones, len := route.Dst.Mask.Size()
+		if (ones == 32 && len == 32) ||
+			(ones == 128 && len == 128) {
+			// if this is needed for the gateway we fail later
+			// otherwise just log the error
+			err := delAddRoute(oldLink, newLink, route)
+			if err != nil {
+				klog.Errorf("Skipping route %v: %v", route, err)
+			}
```

Comment:
I left this for clarification, I just kept the change for discussion because I don´t have clear what is the correct behaviour.
The comment says that the first loop is to handle only a special case in GCE with /32 addresses, if that is then we should manage here only the special case. If not, then we should change the comment to say we install whatever route that matches the filter

Reply:
this is no longer needed